changelog/9061.improvement.rst

```diff
@@ -0,0 +1,15 @@
+Using :func:`pytest.approx` in a boolean context now raises an error hinting at the proper usage.
```

Comment:
Oh just realized, perhaps this should be `breaking`? It will break test suites which are misusing it.

Reply:
I wouldn't consider it breaking, because those test suites are broken already and the documented behavior of `approx()` is unchanged.  But I can see it both ways.

Reply:
Yeah I understand your point, but under our definition it would be a "breaking" change because it will break the test suites, even if the test suites are incorrect, so it is our policy to make it prominently in the CHANGELOG (We have done the same in the past, where we make an incorrect behavior an error but test suites were passing otherwise).

Reply:
Ok, I'll change the file name, one sec...